src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/Network/LinuxNetworkUtilizationParser.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using Microsoft.Extensions.ObjectPool;
+#if !NET8_0_OR_GREATER
+using Microsoft.Shared.StringSplit;
+#endif
+using Microsoft.Shared.Diagnostics;
+using Microsoft.Shared.Pools;
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux.Network;
+
+internal sealed class LinuxNetworkUtilizationParser
+{
+    private static readonly ObjectPool<BufferWriter<char>> _sharedBufferWriterPool = BufferWriterPool.CreateBufferWriterPool<char>();
+
+    /// <remarks>
+    /// File that provide information about currently active TCP_IPv4 connections.
+    /// </remarks>
+    private static readonly FileInfo _tcp = new("/proc/net/tcp");
+
+    /// <remarks>
+    /// File that provide information about currently active TCP_IPv6 connections.
+    /// </remarks>
+    private static readonly FileInfo _tcp6 = new("/proc/net/tcp6");
+
+    private readonly IFileSystem _fileSystem;
+
+    /// <remarks>
+    /// Reads the contents of a file located at _tcp4 and parses it to extract information about the TCP/IP state info on the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv4StateInfo() => GetTcpStateInfo(_tcp);
+
+    /// <remarks>
+    /// Reads the contents of a file located at _tcp6 and parses it to extract information about the TCP/IP state info on the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv6StateInfo() => GetTcpStateInfo(_tcp6);
+
+    public LinuxNetworkUtilizationParser(IFileSystem fileSystem)
+    {
+        _fileSystem = fileSystem;
+    }
+
+    /// <remarks>
+    /// The method is used to read Span data and calculate the TCP state info.
```

Comment:
```suggestion
    /// The method is used to read span data and calculate the TCP state info.
```

What's "span data"? It'd be great to provide a more meaningful description here.

Reply:
Updated